src/glove.jl

```diff
@@ -0,0 +1,83 @@
+                "03d5d7fa28e58762ace4b85fb71fe86a345ef0b5ff39f5390c14869da0fc1970",
+               []),
+               ("glove.840B.300d",
+                "Trained on 840B tokens (Common Crawl), 2.2M vocab, cased. Includes 300d vectors. 2.03 GB download.",
+                "c06db255e65095393609f19a4cfca20bf3a71e20cc53e892aafa490347e3849f",
+                []),
+               ("glove.twitter.27B",
+                "Trained on 2B tweets, 27B tokens, 1.2M vocab, uncased, 25d, 50d, 100d, & 200d vectors, 1.42 GB download.",
+                "792af52f795d1a32c9842a3240f5f3fe5e941a8ff6df5eb0f9d668092ebc019c",
+                ["25d", "50d", "100d", "200d"])]
+    for (depname, description, sha, dims) in vectors
+        register(DataDep(depname,
+                         """
+                         Pretrained GloVe word embeddings.
+                         Website: https://nlp.stanford.edu/projects/glove/
+                         Author: Jeffrey Pennington, Richard Socher, Christopher D. Manning 
+                         Year: 2014
+                         Licence: Open Data Commons Public Domain Dedication and License (PDDL)
+
+                         Paper:
+                             Jeffrey Pennington, Richard Socher, Christopher D. Manning. GloVe: Global Vectors for Word Representation. In Proceedings of EMNLP, 2014.
+
+                         $description
+                         """,
+                         "https://nlp.stanford.edu/data/$(depname).zip",
+                         sha,
+                         post_fetch_method = unpack))
+        if length(dims) >= 1
+            append!(language_files(GloVe{:en}), ["$(depname)/$(depname).$(dim).txt" for dim in dims])
+        else
+            push!(language_files(GloVe{:en}), "$depname/$depname.txt")
+        end
+    end
+end
+
+function glove_max_size(filename)
+    if occursin("glove.6B", filename)
+        return 40000
+    elseif occursin("glove.42B.300d", filename)
+        return 1900000
+    elseif occursin("glove.840B.300d", filename)
+        return 2200000
+    elseif occursin("glove.twitter.27B", filename)
+        return 1200000
+    else
+        error("unknown GloVe file '$filename'")
+    end
+end
+
+_glove_dim(filename) = Meta.parse(match(r"([0-9]+)d.txt$", filename).captures[1])
+
+function _load_embeddings(::Type{<:GloVe}, embedding_file, max_vocab_size, keep_words)
+    local LL, indexed_words, index, D
+    D = _glove_dim(embedding_file)
+    max_stored_vocab_size = min(max_vocab_size, glove_max_size(embedding_file))
+    indexed_words = Vector{String}(undef, max_stored_vocab_size)
+    LL = Array{Float32}(undef, D, max_stored_vocab_size)
+    index = 1
+    open(embedding_file) do f
+        for line in eachline(f)
+            xs = split(line)
+            word = xs[1]
+            if length(keep_words) == 0 || (word in keep_words)
+                @inbounds LL[:, index] = Meta.parse.(xs[2:end])
```

Comment:
Not `Meta.parse` that is for parsing julia code.
`parse.(Ref(Float32), xs[2:end])`
for parsing Floats

Reply:
Oh good, I was hoping for something like this! I wasn't too happy using Meta.parse but didn't know if there was something better.